### DISCUSSION TITLE
Filter and ProjectionExpression on nested attribute

### DISCUSSION THREAD

**Discussion by Jdevvv** [sensedeep/dynamodb-onetable#24] [Q&A] [answered]
Hey,

First of all, congratulations for all this work :pray:

I will soon test your lib to eventually switch on it and I was wondering if it was possible to filter and selectively return attributes on nested attributes (like on Array or Object) ? (I didn't find it in the doc)

**Comment by mobsense** [comment]
Thank you. 

Looks like it is missing in the Model params table, but is mentioned in the doc: search for 'params.fields'.

We'll update the doc.

params.fields = [ fields to return ]

This will set the ProjectionExpression to the given fields.

**Reply by Jdevvv** [reply]
Thanks for your response!

I mean for the params.fields, is it possible to only query on some fields in nested attribute ?

Like for 
`
const obj = {
  firstname: "John",
  lastname: "Doe",
  age: "21"
};
`

something like
`
params.fields = [ obj.lastname, obj.age ];  
//no obj.firstname
`

**Reply by Jdevvv** [reply]
And for Filters :
`
params.where = '(${obj.firstname} = {John})'
`

**Comment by mobsense** [comment, marked as answer]
You use param.fields like:

fields: ['lastname', 'age']

The param.fields is converted into a ProjectionExpression.

For params.where, those are used to create ConditionalExpressions for update/delete or FilterExpressions for find/scan.

I think you can use nested attributes on expressions. Can you test and let us know how you go.

See:

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.Attributes.html
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ProjectionExpressions.html
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ExpressionAttributeNames.html
